File: assets/code-variants/ch12/09.java
@Test
public void testDisplayTransactionResults() {
transactionProcessor.displayTransactionResults(newUserWithBalance(LOW_BALANCE_THRESHOLD.plus(dollars(2))), new Transaction("Some Item", dollars(3)));
    assertThat(ui.getText()).contains("You bought a Some Item");
    assertThat(ui.getText()).contains("your balance is low");
}

File: assets/code-variants/ch12/09.py
def test_display_transaction_results():
    transaction_processor.display_transaction_results(
        new_user_with_balance(LOW_BALANCE_THRESHOLD + dollars(2)), Transaction("Some Item", dollars(3)))
    assert "You bought a Some Item" in ui.get_text()
    assert "your balance is low" in ui.get_text()
